org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/RuntimeClasspathEntry.java

```diff
@@ -352,8 +352,9 @@ public IResource getResource() {
 	protected IResource getResource(IPath path) {
 		if (path != null) {
 			IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
-			if (getType() == PROJECT) {
-				// project entry should always have a workspace relative path, try the fastest lookup first
+			if (getType() == PROJECT || path.getDevice() == null) {
+				// Project entry should always have a workspace relative path, try the fastest lookup first.
+				// Also entry without device has a great chance to find the answer quickly in the workspace.
 				IResource member = root.findMember(path);
```

Comment:
What if it returns a folder? Have you tried that case?
Should we add a check for IContainer type and call the findMember again?

Reply:
Also note, that org.eclipse.jdt.internal.core.JavaModel.getTarget(IPath, boolean) is used for the same purpose in for example org.eclipse.jdt.internal.core.ClasspathEntry.toString() and always checks the workspace first - independent of the type.